app/models/user.rb

```diff
@@ -10,6 +10,10 @@ class User < ApplicationRecord
   has_many :categories, dependent: :destroy
 
   def balance
-    transactions.sum(:amount).to_i
+    transactions.balance_transactions.sum(:amount).to_i
+  end
+
+  def free_balance
+    balance - categories.reduce(0) { |sum, category| sum + category.amount }
```

Comment:
@sinpai why you don't use `sum`, how in `#balance` method? 
``` ruby
def free_balance
  balance - categories.sum(:amount)
end
```
I think it should work.